package.json

```
@@ -51,9 +51,35 @@
   },
   "lint-staged": {
     "*.php": [
-      "composer run-script lint",
       "composer run-script phpstan"
     ],
+    "./*.php": [
+      "composer run-script lint"
+    ],
+    "./tests/*.php": [
+      "composer run-script lint"
+    ],
+    "./{includes,tests/includes,tests/testdata,tests/utils}/**/*.php": [
+      "composer run-script lint"
+    ],
+    "./{plugins/auto-sizes,tests/plugins/auto-sizes}/**/*.php": [
+      "composer run-script lint:auto-sizes"
+    ],
+    "./{plugins/dominant-color-images,tests/plugins/dominant-color-images}/**/*.php": [
+      "composer run-script lint:dominant-color-images"
+    ],
+    "./{plugins/embed-optimizer,tests/plugins/embed-optimizer}/**/*.php": [
+      "composer run-script lint:embed-optimizer"
+    ],
+    "./{plugins/optimization-detective,tests/plugins/optimization-detective}/**/*.php": [
+      "composer run-script lint:optimization-detective"
+    ],
+    "./{plugins/speculation-rules,tests/plugins/speculation-rules}/**/*.php": [
+      "composer run-script lint:speculation-rules"
+    ],
+    "./{plugins/webp-uploads,tests/plugins/webp-uploads}/**/*.php": [
```

These can be simplified once this is merged: #1065.

```suggestion
    "./plugins/auto-sizes/**/*.php": [
      "composer run-script lint:auto-sizes"
    ],
    "./plugins/dominant-color-images/**/*.php": [
      "composer run-script lint:dominant-color-images"
    ],
    "./plugins/embed-optimizer/**/*.php": [
      "composer run-script lint:embed-optimizer"
    ],
    "./plugins/optimization-detective/**/*.php": [
      "composer run-script lint:optimization-detective"
    ],
    "./plugins/speculation-rules/**/*.php": [
      "composer run-script lint:speculation-rules"
    ],
    "./plugins/webp-uploads/**/*.php": [
```